File: android/app/src/debug/java/com/laneshadow/sandbox/stories/LSSurfaceStories.kt
package com.laneshadow.sandbox.stories

import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.padding
import androidx.compose.runtime.Composable
import androidx.compose.ui.Modifier
import com.laneshadow.theme.LaneShadowTheme
import com.laneshadow.theme.LocalLaneShadowTheme
import com.laneshadow.ui.atoms.AccentColor
import com.laneshadow.ui.atoms.GlassVariant
import com.laneshadow.ui.atoms.LSCard
import com.laneshadow.ui.atoms.LSGlassPanel
import com.laneshadow.ui.atoms.LSPanel
import com.laneshadow.ui.atoms.LSText
import com.laneshadow.ui.atoms.TypographyVariant
import com.nativesandbox.model.ComponentTier
import com.nativesandbox.model.Story

object LSSurfaceStories {
    val all: List<Story> = listOf(
        Story(
            id = "atoms.card.default",
            tier = ComponentTier.Atom,
            component = "LSCard",
            name = "Default",
            summary = "Elevated card surface using card background, large radius, and elevation.",
            content = { SurfaceStoryFrame { LSCard { SurfaceStoryCopy("Card surface") } } },
        ),
        Story(
            id = "atoms.panel.default",
            tier = ComponentTier.Atom,
            component = "LSPanel",
            name = "Default",
            summary = "Flat panel surface for grouped content within the primary canvas.",
            content = { SurfaceStoryFrame { LSPanel { SurfaceStoryCopy("Panel surface") } } },
        ),
        Story(
            id = "atoms.glasspanel.chrome",
            tier = ComponentTier.Atom,
            component = "LSGlassPanel",
            name = "Chrome",
            summary = "Ambient translucent chrome panel with blur-backed glass styling.",
            content = {
                SurfaceStoryFrame {
                    LSGlassPanel(variant = GlassVariant.Chrome) {
                        SurfaceStoryCopy("Glass chrome")
                    }
                }
            },
        ),
        Story(
            id = "atoms.glasspanel.callout.signal",
            tier = ComponentTier.Atom,
            component = "LSGlassPanel",
            name = "Callout Signal",
            summary = "Glass callout with signal stripe for navigational emphasis.",
            content = {
                SurfaceStoryFrame {
                    LSGlassPanel(variant = GlassVariant.Callout(AccentColor.Signal)) {
                        SurfaceStoryCopy("Signal callout")
                    }
                }
            },
        ),
        Story(
            id = "atoms.glasspanel.callout.warning",
            tier = ComponentTier.Atom,
            component = "LSGlassPanel",
            name = "Callout Warning",
            summary = "Glass callout with warning stripe for degraded-state messaging.",
            content = {
                SurfaceStoryFrame {
                    LSGlassPanel(variant = GlassVariant.Callout(AccentColor.Warning)) {
                        SurfaceStoryCopy("Warning callout")
                    }
                }
            },
        ),
    )
}

@Composable
private fun SurfaceStoryFrame(content: @Composable () -> Unit) {
    LaneShadowTheme {
        val theme = LocalLaneShadowTheme.current
        Column(
            modifier = Modifier.padding(theme.space.xl),
            verticalArrangement = Arrangement.spacedBy(theme.space.md),
        ) {
            content()
        }
    }
}

@Composable
private fun SurfaceStoryCopy(label: String) {
    LSText(
        text = label,
        variant = TypographyVariant.Ui.Body.Md,
    )
}

File: ios/LaneShadow/Sandbox/Stories/LSSurfaceStories.swift
import LaneShadowTheme
import NativeSandbox
import SwiftUI

@MainActor
enum LSSurfaceStories {
    static let all: [Story] = [
        Story(
            id: "atoms.card.default",
            tier: .atom,
            component: "LSCard",
            name: "Card Default",
            summary: "Elevated card with the default surface tokens."
        ) { _ in
            SurfaceCardDefaultStory()
        },
        Story(
            id: "atoms.card.with-content",
            tier: .atom,
            component: "LSCard",
            name: "Card With Content",
            summary: "Elevated card with richer content composition."
        ) { _ in
            SurfaceCardContentStory()
        },
        Story(
            id: "atoms.panel.default",
            tier: .atom,
            component: "LSPanel",
            name: "Panel Default",
            summary: "Flat panel surface with compact padding."
        ) { _ in
            SurfacePanelDefaultStory()
        },
        Story(
            id: "atoms.panel.nested",
            tier: .atom,
            component: "LSPanel",
            name: "Panel Nested",
            summary: "Nested panel composition inside a card."
        ) { _ in
            SurfacePanelNestedStory()
        },
        Story(
            id: "atoms.glasspanel.chrome",
            tier: .atom,
            component: "LSGlassPanel",
            name: "GlassPanel Chrome",
            summary: "Backdrop-blurred chrome surface."
        ) { _ in
            SurfaceGlassChromeStory()
        },
        Story(
            id: "atoms.glasspanel.callout-signal",
            tier: .atom,
            component: "LSGlassPanel",
            name: "GlassPanel Callout Signal",
            summary: "Glass callout with the signal accent stripe."
        ) { _ in
            SurfaceGlassCalloutStory(accent: .signal)
        },
        Story(
            id: "atoms.glasspanel.callout-warning",
            tier: .atom,
            component: "LSGlassPanel",
            name: "GlassPanel Callout Warning",
            summary: "Glass callout with the warning accent stripe."
        ) { _ in
            SurfaceGlassCalloutStory(accent: .warning)
        },
    ]
}

private struct SurfaceCardDefaultStory: View {
    var body: some View {
        LSCard {
            LSText("Ride Summary", variant: .title.md)
        }
        .padding(Theme.shared.space.lg)
    }
}

private struct SurfaceCardContentStory: View {
    @Environment(\.theme) private var theme

    var body: some View {
        LSCard {
            VStack(alignment: .leading, spacing: theme.space.sm) {
                LSText("Favorite Route", variant: .title.md)
                LSText("64 mi  1h 42m  Scenic river loop", variant: .body.md, color: .secondary)
            }
        }
        .padding(theme.space.lg)
    }
}

private struct SurfacePanelDefaultStory: View {
    var body: some View {
        LSPanel {
            LSText("Panel Default", variant: .body.md)
        }
        .padding(Theme.shared.space.lg)
    }
}

private struct SurfacePanelNestedStory: View {
    @Environment(\.theme) private var theme

    var body: some View {
        LSCard {
            VStack(alignment: .leading, spacing: theme.space.md) {
                LSText("Nested Surface", variant: .title.md)
                LSPanel {
                    LSText("Panels stack cleanly inside cards.", variant: .body.md, color: .secondary)
                }
            }
        }
        .padding(theme.space.lg)
    }
}

private struct SurfaceGlassChromeStory: View {
    @Environment(\.theme) private var theme

    var body: some View {
        ZStack {
            RoundedRectangle(cornerRadius: theme.radius.xl, style: .continuous)
                .fill(theme.colors.surfaceVariant.default)
                .frame(height: 180)

            LSGlassPanel {
                VStack(alignment: .leading, spacing: theme.space.sm) {
                    LSText("Chrome Surface", variant: .title.md)
                    LSText("Used for HUD and overlay chrome.", variant: .body.md, color: .secondary)
                }
            }
            .padding(theme.space.lg)
        }
        .padding(theme.space.lg)
    }
}

private struct SurfaceGlassCalloutStory: View {
    @Environment(\.theme) private var theme

    let accent: AccentColor

    var body: some View {
        ZStack {
            RoundedRectangle(cornerRadius: theme.radius.xl, style: .continuous)
                .fill(theme.colors.surfaceVariant.default)
                .frame(height: 180)

            LSGlassPanel(variant: .callout(accent: accent)) {
                VStack(alignment: .leading, spacing: theme.space.sm) {
                    LSText(title, variant: .title.md)
                    LSText(message, variant: .body.md, color: .secondary)
                }
            }
            .padding(theme.space.lg)
        }
        .padding(theme.space.lg)
    }

    private var title: String {
        switch accent {
        case .signal:
            "Signal Callout"
        case .warning:
            "Warning Callout"
        }
    }

    private var message: String {
        switch accent {
        case .signal:
            "Clear skies. Best conditions for a long ride."
        case .warning:
            "Watch the canyon gusts after 4 PM."
        }
    }
}
